### PR TITLE
Prevent Conversion 2 from generating illegal types

### PR DESCRIPTION
**Pull request by dot-Comfey**
https://www.smogon.com/forums/threads/conversion2-allows-the-fairy-type-to-exist-in-pre-fairy-gens.3754371/

### CODE REVIEW

**Comment by larry-the-table-guy** [comment]
Should `gen8linked` have a similar fix? https://github.com/smogon/pokemon-showdown/blob/master/data/mods/gen8linked/moves.ts#L380

**Comment by dot-Comfey** [comment]
I saw that too, and I'm not sure if it'd be necessary because the check should always pass in Gen 8 (the only isNonstandard type is Stellar which should never be generated by Conversion 2 anyways), but at the same time it would provide parity.

**Comment by KrisXV** [comment]
Closing since I went ahead and did this and fixed other instances like this in https://github.com/smogon/pokemon-showdown/commit/63ca4e2c8144320e2c28584946bc37570b29da89

File: data/moves.ts [hunk 1]
```diff
@@ -2909,6 +2909,7 @@ export const Moves: import('../sim/dex-moves').MoveDataTable = {
 			const possibleTypes = [];
 			const attackType = target.lastMoveUsed.type;
 			for (const type of this.dex.types.all()) {
```
Comment (Zarel) [inline data/moves.ts:2911]: Would this be more readable?

```ts
const possibleTypes = this.dex.types.all().filter(type => (
  !type.isNonstandard && !source.hasType(type.name) && [2, 3].includes(type.damageTaken[attackType])
))
```

**Pull request closed without merging**